fluent/protocol/transport.go

```diff
@@ -130,20 +133,8 @@ func (e EntryList) Equal(e2 EntryList) bool {
 		for _, eb := range second {
 			if ea.Timestamp.Equal(eb.Timestamp.Time) {
 				// Timestamps equal, check the record
-				if len(ea.Record) == len(eb.Record) {
-					// This only works if we have the same number of kv pairs in each record
-					for k, v := range ea.Record {
-						if eb.Record[k] == v {
-							// KV match, so delete key from each record
-							delete(ea.Record, k)
-							delete(eb.Record, k)
-						}
-					}
-
-					if len(ea.Record) == 0 && len(eb.Record) == 0 {
-						// No more keys left means everything matched
-						matches++
-					}
+				if reflect.DeepEqual(ea.Record, eb.Record) {
```

Comment:
Possible improvement in performance if we can implement our own logic to distinguish records diff!